hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -175,8 +183,17 @@ public BlockRecordInfo blockRecordInfo() {
      */
     @Override
     public long newEntityNum() {
-        final var writableStoreFactory = new WritableStoreFactory(stack, EntityIdService.NAME);
-        return writableStoreFactory.getStore(WritableEntityIdStore.class).incrementAndGet();
+        final var entityIdsFactory = new WritableStoreFactory(stack, EntityIdService.NAME);
+        return entityIdsFactory.getStore(WritableEntityIdStore.class).incrementAndGet();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long peekAtNewEntityNum() {
+        final var entityIdsFactory = new WritableStoreFactory(stack, EntityIdService.NAME);
```

Comment:
Nit: we should probably introduce a private helper method that gets the `WritableEntityIdStore`